lib/metasploit/framework/login_scanner/snmp.rb

```diff
@@ -191,10 +202,30 @@ def process_logins(opts = {})
           process_responses(1.0)
         end
 
+        def recv_wrapper(sock, max_size, timeout)
+          res = nil
+          if protocol == 'udp'
+            res = sock.recvfrom(max_size, timeout)
+          end
+
+          if protocol == 'tcp'
+            ready = ::IO.select([sock], nil, nil, timeout)
+            if ready
+              res = sock.recv_nonblock(max_size)
+              # Put into an array to mimic recvfrom
+              res = [res, host, port]
+            else
+              # print("timeout\n")
+            end
+          end
+
+          return res
+        end
+
```

Comment:
It looks like this could be cleaned up just a touch.

```suggestion
        def recv_wrapper(sock, max_size, timeout)
          res = nil
          if protocol == 'udp'
            res = sock.recvfrom(max_size, timeout)
          elsif protocol == 'tcp'
            ready = ::IO.select([sock], nil, nil, timeout)
            if ready
              res = sock.recv_nonblock(max_size)
              # Put into an array to mimic recvfrom
              res = [res, host, port]
            end
          end

          res
        end

```

It also looks like the only place this is used just wants the response. You could normalize on just returning data when it's received and `nil` otherwise.

Reply:
Ok - I will adjust